tests/integration/ember-concurrency-ts-test.ts

```diff
@@ -96,4 +103,95 @@ module('Integration | ember-concurrency-ts', function(hooks) {
     assert.dom('#value').hasText('Done!');
     assert.dom('#resolved').hasText('Wow!');
   });
+
+  test('it works for encapsulated tasks', async function(assert) {
+    let { promise, resolve } = defer<string>();
+
+    class MyComponent extends Component {
+      lastValue: string | null = null;
+
+      @task myTask = {
+        resolved: '',
+        *perform(arg: string): TaskGenerator<string> {
+          set(this, 'resolved', yield promise);
```

Comment:
Should this check for these things https://github.com/machty/ember-concurrency/blob/d5675eb1f742e8f7bb4698e49bd8063a3cc2ba2e/tests/types/ember-concurrency-test.ts#L2364-L2366

Reply:
Good idea since the `yield` doesn't gives us type safety.

Reply:
I've also added this to #1 for good measure.